zkdb/src/storage-engine/ipfs.ts

```diff
@@ -291,6 +291,30 @@ export class StorageEngineIPFS implements TIPFSFileSystem, TIPFSFileIndex {
     this.pathBase = path;
   }
 
+  // Try to resolve IPNS entry to CID
+  public async tryResolve() {
+    if (typeof (await this.resolve()) === 'undefined') {
+      await this.initializeMetadata();
+    }
+  }
+
+  // Initial missing data
+  public async initializeMetadata() {
+    if (!fs.existsSync(this.pathStorageMetadata)) {
+      const docMetadata: { [key: string]: string } = {};
+      const metadataContent = BSON.serialize(docMetadata);
+      fs.writeFileSync(this.pathStorageMetadata, metadataContent);
+      const cid = await this.unixFs.addBytes(metadataContent);
```

Comment:
Could we move out repeated blocks of code?

Reply:
We can let the refactor to next stage, for now just keep it work. The interface still unstable so I won't put my effort to refactor the thing that ins't unstable.